src/Examples/VGG.cs

```diff
@@ -54,7 +54,7 @@ public VGG(string name, int numClasses, Device device = null) : base(name)
 
             RegisterComponents();
 
-            if (device != null && device.type == DeviceType.CUDA)
```

Comment:
You can't remove the null check as long as 'null' is a possible value, which it is given the default value in the signature. This goes for all these changes.